rllib/agents/a3c/a3c_torch_policy.py

```diff
@@ -45,14 +45,12 @@ def add_advantages(policy,
     else:
         last_r = policy._value(sample_batch[SampleBatch.NEXT_OBS][-1])
 
-    return compute_advantages(sample_batch, last_r, policy.config["gamma"],
-                              policy.config["lambda"],
-                              policy.config["use_gae"],
-                              policy.config["use_critic"])
+    return compute_advantages(
+        sample_batch, last_r, policy.config["gamma"], policy.config["lambda"],
+        policy.config["use_gae"], policy.config["use_critic"])
 
 
-def model_value_predictions(policy, input_dict, state_batches, model,
-                            action_dist):
+def model_value_predictions(policy, input_dict, state_batches, model):
```

Comment:
> This is a pretty big PR with lots of changes. Logistically, can we break this down into a series of smaller PRs for review?
> 
> 1. Add new param noise exploration interface.
ok, will split this into 2.

> 2. PR to port DQN/SimpleQ.
see above

> 3. PR to port remaining algos.
TODO later (DDPG)

> 4. Other refactorings.
> 
> Design wise, I also think we can get away with a much more minimal interface for supporting exploration:
> 
> 1. On episode start, on episode end hooks -> used to add and remove noise from the model.
> 2. Before compute action hook -> used to implement dynamic changing of the `exploration` argument in case the user needs to call compute_actions() outside an episode trajectory.

Yes, great idea, for parameter noise that would actually work, b/c we are already storing the noisy-state of the net and could only revert (before the next forward pass, if anything changed). We could get rid of after_forward_pass this way.

> That would only require a single change to the Policy classes to call the before compute action hook, without introducing new abstractions or changing existing ones.